extensions/nextdns/src/components/list-item.tsx

```diff
@@ -0,0 +1,106 @@
+import { Action, ActionPanel, Color, Icon, Keyboard, List, showToast, Toast } from "@raycast/api";
+import { removeDomain, toggleDomain } from "../libs/api";
+import { DomainListItem, Mutate } from "../types";
+import AddDomain from "./add-domain";
+import { getIconById } from "../libs/utils";
+
+export function ListItem(props: { domainItem: DomainListItem; mutate: Mutate }) {
+  const { domainItem, mutate } = props;
+
+  return (
+    <List.Item
+      title={`*.${domainItem.id}`}
+      icon={getIconById(domainItem.id)}
+      accessories={[
+        {
+          tag: domainItem.active
+            ? { color: Color.Green, value: "Activated" }
+            : { color: Color.Red, value: "Deactivated" },
+        },
+      ]}
+      actions={<Actions item={domainItem} mutate={mutate} />}
+    />
+  );
+}
+
+function Actions({ item, mutate }: { item: DomainListItem; mutate: Mutate }) {
+  async function toggle(item: DomainListItem) {
+    const newStatus = !item.active;
+    const { id, type } = item;
+
+    const toast = await showToast({
+      style: Toast.Style.Animated,
+      title: newStatus ? "Activating domain" : "Deactivating domain",
+      message: id,
+    });
+    try {
+      await mutate(toggleDomain({ element: { id, type, active: newStatus } }), {
+        optimisticUpdate(data) {
+          const { result, profileName } = data;
+          const index = result.findIndex((item) => item.id === id);
+          result[index].active = newStatus;
+          return { result, profileName };
+        },
+      });
+      toast.style = Toast.Style.Success;
+      toast.title = newStatus ? "Activated domain" : "Deactivated domain";
+    } catch (error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = `Could not ${newStatus ? "activate" : "deactivate"} domain`;
+    }
+  }
+
+  async function removeItem(element: DomainListItem) {
+    const toast = await showToast({
+      style: Toast.Style.Animated,
+      title: "Removing domain",
+      message: element.id,
+    });
+
+    try {
+      await mutate(removeDomain({ element }), {
+        optimisticUpdate(data) {
+          const list = data.result;
+          const index = list.findIndex((item) => item.id === element.id);
+
+          if (index !== -1) {
+            list.splice(index, 1);
+          }
+
+          return { ...data, result: list };
+        },
+      });
+
+      toast.style = Toast.Style.Success;
+      toast.title = "Removed domain";
+    } catch (error) {
+      toast.style = Toast.Style.Failure;
+      toast.title = `Could not remove domain`;
+    }
+  }
+
+  return (
+    <ActionPanel>
+      <ActionPanel.Section title={`Manage ${item.id}`}>
+        <Action
+          title={`${item.active ? "Deactivate" : "Activate"} Domain`}
+          icon={item.active ? Icon.XMarkCircle : Icon.CheckCircle}
+          onAction={() => toggle(item)}
```

Comment:
Should we not prompt for confirmation in case someone is deactivating a domain? Seems like a destructive action (we can use destructive style too in that case).